extensions/amp-apester-media/0.1/monetization/companion/video.js

```diff
@@ -21,29 +20,56 @@ export function handleCompanionVideo(media, apesterElement, consentObj) {
   const position = getCompanionPosition(
     /**@type {!JsonObject}*/ (videoSettings)
   );
-
   if (
-    !companionCampaignOptions ||
     !videoSettings ||
-    !videoSettings['enabled'] ||
-    videoSettings['provider'] !== ALLOWED_AD_PROVIDER ||
+    !videoSettings.enabled ||
     !position ||
     position === 'floating'
   ) {
     return;
   }
-  const macros = getSrMacros(
-    media,
-    companionCampaignOptions['companionCampaignId'],
-    apesterElement,
-    consentObj
-  );
-  addCompanionSrElement(
-    videoSettings['videoTag'],
-    position,
-    /** @type {!JsonObject} */ (macros),
-    apesterElement
-  );
+  setStyles(apesterElement, {
+    'height': px(parseInt(apesterElement.getAttribute('height'), 10)),
```

Comment:
Ditto here.

Reply:
I can not put ad height logic here because it will be some time delay and user see the shift. On the first case we get height from the attribute and when the unit is ready we correlate the size with real height on the "/monetization/companion/video.js". I've tried to put logic here before but use will be see shift in this case.

Reply:
If I move height ad block logic here we'll get sift. Take a look on the video with your suggestion:

https://user-images.githubusercontent.com/1662080/144014294-b29fc3c8-ece0-442c-9bb0-342ef42c3d1f.mov

Reply:
This is my solution with height logic in the "amp-apester-media.js".

https://user-images.githubusercontent.com/1662080/144014456-98418726-87ba-4883-9230-7a9392ed64f3.mov

Reply:
@jridgewell please take a look on my comments and video screens. Thank you for your help and review.

Reply:
@golovatuy I think justin's point is that, rather than the publisher doing `<amp-apester-media height=100`, why not just `<amp-apester-media height=200>` (which includes your survey unit?

Reply:
Exactly. The two uses here seem to be `<amp-apester-media height=100 extra-height=20>`, why is this not just `height=120`. That avoids the shift entirely in all circumstances.

Reply:
@jridgewell @powerivq Ad campaign is setting up by user independently from creating media unit (amp-apester-media), we can not mix them. There is another issue with old unit which already published. Also "height" attribute is responsible only for exactly media height (slides with images and/or video), ad block is placed in the container outside of it. Media and ad block are different entities, which logic could not be mixed at this moment.

Reply:
I don't understand. You're reading an attribute from an element, which means the publisher is setting it in HTML. If we can read that value in HTML, why isn't the height correctly set just via the HTML. Why does JS need to execute in order to update this style?

Reply:
@jridgewell You're right. Just using extra in height is much better and more clean. I fixed it in the last commit.